src/client.rs

```diff
@@ -43,6 +45,50 @@ use crate::{
     AstarteAggregate, Error, Interface,
 };
 
+/// Possible errors returned while handling connection messages
+#[non_exhaustive]
+#[derive(thiserror::Error, Debug)]
+pub enum RecvError {
+    /// Connection error, either gRPC or MQTT
+    #[error("connection error, {0:?}")]
+    Connection(#[source] crate::transport::ConnectionRecvError),
+
+    /// Couldn't parse the mapping path.
+    #[error("invalid mapping path '{}'", .0.path())]
+    InvalidEndpoint(#[from] MappingError),
+
+    /// Couldn't find an interface with the given name.
+    #[error("couldn't find interface '{name}'")]
+    InterfaceNotFound {
+        /// Name of the missing interface.
+        name: String,
+    },
+
+    /// Couldn't find missing mapping in the interface.
+    #[error("couldn't find mapping {mapping} in interface {interface}")]
+    MappingNotFound {
+        /// Name of the interface.
+        interface: String,
+        /// Path of the missing mapping.
+        mapping: String,
+    },
+
+    /// Invalid aggregation between the interface and the data.
+    #[error("invalid aggregation, expected {exp} but got {got}")]
+    Aggregation {
```

Comment:
Does it represent all interface types, or only the Datastream (aggregation individual/object) type?

Reply:
It represents only the Individual/Object datastream type.

Reply:
Since sometimes the properties are handled as Individual type, If there is an error on `property` side will an Aggregation error be used, or should we add an specific error?

Reply:
I added some context to the `Aggregate` error variant, so now the distinction should be clearer